config/v1/0000_10_config-operator_01_infrastructure-CustomNoUpgrade.crd.yaml

```diff
@@ -466,6 +466,55 @@ spec:
                     aws:
                       description: AWS contains settings specific to the Amazon Web Services infrastructure provider.
                       properties:
+                        dnsConfig:
+                          description: dnsConfig contains information about API and API-Int Load Balancer and DNS configuration.
+                          properties:
+                            clusterProvided:
+                              description: clusterProvided contains information about the OpenShift cluster created LoadBalancers.
+                              properties:
+                                apiIntLB:
+                                  description: apiIntLB contains configuration for the API-Int Load Balancer.
+                                  properties:
+                                    name:
+                                      description: name is the name of the LB created by the user.
+                                      type: string
+                                  type: object
+                                apiLB:
+                                  description: apiLB contains configuration for the API Load Balancer
+                                  properties:
+                                    name:
+                                      description: name is the name of the LB created by the user.
+                                      type: string
+                                  type: object
+                              type: object
+                            provider:
+                              default: ClusterProvided
+                              description: provider determines which DNS solution is in use for this cluster. When the user has pre-created the Load Balancers for API and API-Int and the the DNS solution has already been configured, this field is set to "UserProvided". When the cluster is resposible for its own DNS configuration and LB creation, this field is set to "ClusterProvided".
+                              enum:
+                                - UserProvided
+                                - ClusterProvided
```

Comment:
When Load Balancers are created by the cluster (which is the default case), the installer could set the 'provider` to "ClusterProvided" and populate the values for the LB Names. But, there is no real use for it.

So, what if the enum have 1 value i.e "USerProvided" ?

Reply:
Interesting idea, so that would be along the lines of `When omitted, this means no opinion and the platform is left to choose a reasonable default` that we have elsewhere. Typically though that information is normally tagged to one of the enum values, eg, in this case, `The default is ClusterProvided` would be the next sentence.

IMO I think it's better to have a value though. Since this is a configuration API we like to have all options populated/displayed. I think it's better to have a value that explains it's ClusterProvided rather than just the empty string

Reply:
Ok. Let us keep it the way it is. It also has the advantage of the default behavior being less ambiguous. Comment updated to reflect that.